File: 3075: maximizeHappinessOfSelectedChildren.cs
public class Solution {
    public long MaximumHappinessSum(int[] happiness, int k) {
        
        Array.Sort(happiness);
        long res = 0;
        int picked = 0;
        int n = happiness.Length;
        while(n > 0 && k > 0){
            int curr = happiness[n-1-picked];
            if((curr - picked) <= 0){

                return res;
            }
            res += curr - picked;
            picked++;
            k--;
        }
        return res;
    }
}

File: 3075: maximizeHappinessOfSelectedChildren.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        res = 0
        picked = 0
        happiness.sort()
        while happiness and k > 0:
            curr = happiness.pop()
            if (curr - picked) <= 0:
                return res
            res += curr - picked
            picked += 1
            k -= 1
        return res